### ISSUE TITLE
Is it Glibc dependent? Can it be statically linked?

### ISSUE THREAD

**Issue by jsav0** [projectdiscovery/httpx#118]
Unable to run the compiled binaries on musl systems (alpine, voidlinux-musl, and others)
```
$ ldd httpx
/lib64/ld-linux-x86-64.so.2 (0x7f5654308000)
libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7f5654308000)
libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7f5654308000)
```

**Comment by Mzack9999** [comment]
@jsav0 - Thanks for opening this issue. Actually it's compiled with cgo support, as we use some low level system calls. It seems that goreleaser doesn't support compiling against musl automatically. Do you have any suggestion about how this can be automated? Maybe a docker multi-stage build?

**Comment by Mzack9999** [comment]
Fixed in #172
